ndcube/ndcube_sequence.py

```diff
@@ -84,14 +84,21 @@ def cube_like_world_axis_physical_types(self):
         return self.data[0].world_axis_physical_types
 
     def __getitem__(self, item):
+
         if isinstance(item, numbers.Integral):
             return self.data[item]
-        elif isinstance(item, slice):
+        else:
             result = copy.deepcopy(self)
-            result.data = self.data[item]
+            if isinstance(item, slice):
+                result.data = self.data[item]
+            else:
+                if isinstance(item[0], numbers.Integral):
+                    result = result.data[item[0]][item[1:]]
+                elif isinstance(item[0], slice) and item[0].stop - item[0].start == 1:
```

Comment:
Just looked at this again.  I don't think this line and the one below are needed at all?  The line after the ```else``` will do the same job in all remaining cases, I'm pretty sure?  If so, that'll also get rid of the test failure.

Reply:
yeah its needed, as if you just do the list comprehension on the `NDCube` rather on the list of NDCubes it then loops through the arrays in the NDCube if that makes sense

Reply:
i.e. as in for this `result.data = [cube[item[1:]] for cube in result.data[item[0]]]` to work you need result.data[item[0]] to be a list of NDCubes rather than an NDCube. but if item[0] is a 0, or slice(0, 1) this doesn't work ... the way it is currently implemented also follows similar logic to the way the utils.sequence.slice_sequence kind of did it without the need of a `SequenceItem`

Reply:
Ah!  You're dead right!

Reply:
```suggestion
                elif isinstance(item[0], slice):
                    start = 0 if item[0].start is None else item[0].start
                    stop = len(self.data) if item[0].stop is None else item[0].stop
                    if stop - start == 1:
                        result.data = [result.data[item[0].start][item[1:]]]
```